### ISSUE TITLE
Problems with MS5xxx.cpp when using Seeduino Zero

### ISSUE THREAD

**Issue by jritz1349** [Schm1tz1/arduino-ms5xxx#8]
When testing the MS5607 with the MS5xxx library it compiles fine with an Arduino uno but when I compile it with a Seeduino Zero I 
get errors:

error: expected primary-expression before 'unsigned' unsigned long D1=0;
error: expected primary-expression before 'unsigned' unsigned long D2=0;

error: lvalue required as left operand of assignment D2=read_adc(MS5xxx_CMD_ADC_D2+MS5xxx_CMD_ADC_4096);
error: lvalue required as left operand of assignment D1=read_adc(MS5xxx_CMD_ADC_D1+MS5xxx_CMD_ADC_4096);

**Comment by Schm1tz1** [comment]
Hi, that's intersting. Can you give the line number of the error? Any special version of arduino or the compiler you are using?
IMO it should be https://github.com/Schm1tz1/arduino-ms5xxx/blob/master/src/MS5611.cpp#L29C18-L29C18 but trying to verify. That line is not best code style but should be valid for Seeduino as far as I can see. Will need to look into this.

**Comment by jritz1349** [comment]
I am using the latest Arduino IDE 2.1.0. I also tried the old Arduino
1.8.9.
I used the MS5xxx.cpp and .h files and it gave me the following errors when
compiling using Seeduino Zero (or XIAO) boards:

C:\Users\johnr\OneDrive\Documents\Arduino\libraries\MS5xxx\src\MS5611.cpp:
In member function 'virtual void MS5611::Readout()':
C:\Users\johnr\OneDrive\Documents\Arduino\libraries\MS5xxx\src\MS5611.cpp:29:2:
error: expected primary-expression before 'unsigned'
  unsigned long D1=0, D2=0;
  ^~~~~~~~
C:\Users\johnr\OneDrive\Documents\Arduino\libraries\MS5xxx\src\MS5611.cpp:35:51:
error: lvalue required as left operand of assignment
  D2=read_adc(MS5xxx_CMD_ADC_D2+MS5xxx_CMD_ADC_4096);
                                                   ^
C:\Users\johnr\OneDrive\Documents\Arduino\libraries\MS5xxx\src\MS5611.cpp:36:51:
error: lvalue required as left operand of assignment
  D1=read_adc(MS5xxx_CMD_ADC_D1+MS5xxx_CMD_ADC_4096);
                                                   ^
C:\Users\johnr\OneDrive\Documents\Arduino\libraries\MS5xxx\src\MS5xxx.cpp:
In member function 'void MS5xxx::Readout()':
C:\Users\johnr\OneDrive\Documents\Arduino\libraries\MS5xxx\src\MS5xxx.cpp:147:2:
error: expected primary-expression before 'unsigned'
  unsigned long D1=0, D2=0;
  ^~~~~~~~
C:\Users\johnr\OneDrive\Documents\Arduino\libraries\MS5xxx\src\MS5xxx.cpp:153:51:
error: lvalue required as left operand of assignment
  D2=read_adc(MS5xxx_CMD_ADC_D2+MS5xxx_CMD_ADC_4096);
                                                   ^
C:\Users\johnr\OneDrive\Documents\Arduino\libraries\MS5xxx\src\MS5xxx.cpp:154:51:
error: lvalue required as left operand of assignment
  D1=read_adc(MS5xxx_CMD_ADC_D1+MS5xxx_CMD_ADC_4096);
                                                   ^
exit status 1
Compilation error: exit status 1

I have attached the .h and .cpp files. It works fine when compiling for a
simple Arduino Uno board.

-John Ritz

**Comment by Schm1tz1** [comment]
Interesting - trying to figure out where's the difference in the compiler / compiler settings for those boards. I cannot find XIAO / Seed* Boards in 1.8.9 or 2.1.1 boards manager. What board configuration are you using and where is it coming from?
Maybe we need to dig a bit deeper.

**Comment by Schm1tz1** [comment]
Gotcha! Installed the boards according to https://wiki.seeedstudio.com/Seeed_Arduino_Boards/ and could reproduce the error. According to the docs, **unsigned long** is a valid type for Seeeduino so I tested that with a minimal example and it works. I found that Seeeduino is using global constants D1 and D2 and compile was crashing due to a conflict. I fixed it locally by renaming like this:
```
void MS5611::Readout() {
	unsigned long l_D1=0, l_D2=0;
	
	double dT;
	double OFF;
	double SENS;

	l_D2=read_adc(MS5xxx_CMD_ADC_D2+MS5xxx_CMD_ADC_4096);
	l_D1=read_adc(MS5xxx_CMD_ADC_D1+MS5xxx_CMD_ADC_4096);
...
```
Will push a patch later so we have a working release online.

**Comment by Schm1tz1** [comment]
Done, PR is #10

**Comment by jritz1349** [comment]
Great! So do I have to install a new MS5xxx.cpp and .h files or just do the
changes that you showed above? The lines in Readout():

unsigned long l_D1=0, l_D2=0;
	
	double dT;
	double OFF;
	double SENS;

	l_D2=read_adc(MS5xxx_CMD_ADC_D2+MS5xxx_CMD_ADC_4096);
	l_D1=read_adc(MS5xxx_CMD_ADC_D1+MS5xxx_CMD_ADC_4096);


-John

**Comment by jritz1349** [comment]
Ok, So everywhere the D1 and D2 is used has to be changed to l_D1 and l_D2
(Is it the letter l not 1, right?).
Because D1 and D2 are used later in computations throughout. Thanks again.

-John

**Comment by Schm1tz1** [comment]
Yes, exactly, all occurrences in the code are renamed. 
You can simply use the new release, it's already fixed in 1.1.1 and available in Arduino librararies. Just update it in your IDE and it should work.

**Comment by jritz1349** [comment]
Yes, I made the updates changing D1,D2 to l_D1,l_D2 and it now compiles
fine with Seeduino Zero. Thank you again!

-John
